src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -42,19 +52,47 @@ internal abstract class AbstractSnippetExpansionClient : ForegroundThreadAffinit
         protected readonly ITextView TextView;
         protected readonly ITextBuffer SubjectBuffer;
 
+        private readonly IEnumerable<Lazy<ArgumentProvider, OrderableLanguageMetadata>> _allArgumentProviders;
+        private ImmutableArray<ArgumentProvider> _argumentProviders;
+
         protected bool indentCaretOnCommit;
         protected int indentDepth;
         protected bool earlyEndExpansionHappened;
 
         internal IVsExpansionSession ExpansionSession;
-
-        public AbstractSnippetExpansionClient(IThreadingContext threadingContext, Guid languageServiceGuid, ITextView textView, ITextBuffer subjectBuffer, IVsEditorAdaptersFactoryService editorAdaptersFactoryService)
+        private bool _preserveSymbols;
+        private ImmutableArray<IMethodSymbol> _symbols;
+        private IMethodSymbol _symbol;
+        private ImmutableDictionary<string, string> _arguments = ImmutableDictionary.Create<string, string>();
+
+        public AbstractSnippetExpansionClient(
+            IThreadingContext threadingContext,
+            Guid languageServiceGuid,
+            ITextView textView,
+            ITextBuffer subjectBuffer,
+            IVsEditorAdaptersFactoryService editorAdaptersFactoryService,
+            IEnumerable<Lazy<ArgumentProvider, OrderableLanguageMetadata>> argumentProviders)
             : base(threadingContext)
         {
             this.LanguageServiceGuid = languageServiceGuid;
             this.TextView = textView;
             this.SubjectBuffer = subjectBuffer;
             this.EditorAdaptersFactoryService = editorAdaptersFactoryService;
+            this._allArgumentProviders = argumentProviders;
+        }
+
+        internal ImmutableDictionary<string, string> Arguments => _arguments;
```

Comment:
any reason gto have the split prop/field, instead of just an auto-prop?

Reply:
Converting to a read/write auto-property.